base/ca/src/main/java/com/netscape/cms/servlet/cert/scep/CRSEnrollment.java

```diff
@@ -2099,19 +2157,27 @@ public CryptoContextException(String s) {
         public CryptoContext()
                 throws CryptoContextException {
             try {
-                KeyGenAlgorithm kga = KeyGenAlgorithm.DES;
-                if (mEncryptionAlgorithm != null && mEncryptionAlgorithm.equals("DES3")) {
+                KeyGenAlgorithm kga;
+                switch(String.valueOf(mEncryptionAlgorithm)) {
+                case "DES3":
                     kga = KeyGenAlgorithm.DES3;
+                    break;
+                case "AES":
+                    kga = KeyGenAlgorithm.AES;
+                    break;
+                default:
+                    kga = KeyGenAlgorithm.DES;
                 }
                 cm = CryptoManager.getInstance();
                 internalToken = cm.getInternalCryptoToken();
-                DESkg = internalToken.getKeyGenerator(kga);
+                keyGen = internalToken.getKeyGenerator(kga);
+                if(kga.equals(KeyGenAlgorithm.AES)) {
+                    keyGen.initialize(128);
```

Comment:
Just for info, this 128 should be fine for now, but in the future might want to make this possible to be larger.

Reply:
According to the most recent [SCEP specs](https://www.rfc-editor.org/rfc/rfc8894.html#section-3.5.2) I have found, the AES capabilities is only for AES128-CBC, other cases are not mentioned so I did not thought to organise the code for different key length/padding combinations. We could consider future update to the specs if you think it is better to do now while we are modifying the code.

Reply:
Sounds good.